packages/frontend/src/scripts/configureDropdowns.ts

```diff
@@ -12,17 +12,61 @@ export function configureDropdowns() {
   for (const dropdown of dropdowns) {
     const button = dropdown.querySelector('.Dropdown-Button')
     if (!button) continue
-    const hiddenItems = dropdown.querySelectorAll('.Dropdown-Item')
-    const transparentItems = dropdown.querySelectorAll(
+    const hiddenItems = dropdown.querySelectorAll<HTMLElement>('.Dropdown-Item')
+    const transparentItems = dropdown.querySelectorAll<HTMLElement>(
       '.Dropdown-Transparent-Item',
     )
 
     button.addEventListener('click', () => {
-      hiddenItems.forEach((item) => item.classList.toggle('hidden'))
-      transparentItems.forEach((item) => item.classList.toggle('opacity-0'))
-      transparentItems.forEach((item) =>
-        item.classList.toggle('pointer-events-none'),
-      )
+      hiddenItems.forEach((item) => {
+        item.classList.toggle('hidden')
+        if (isCentered(item)) {
+          recenter(dropdown, item)
+        }
+      })
+      transparentItems.forEach((item) => {
+        item.classList.toggle('opacity-0')
+        item.classList.toggle('pointer-events-none')
+        if (isCentered(item)) {
+          recenter(dropdown, item)
+        }
+      })
+    })
+    window.addEventListener('resize', () => {
+      hiddenItems.forEach((item) => {
+        if (isCentered(item)) {
+          recenter(dropdown, item)
+        }
+      })
+      transparentItems.forEach((item) => {
+        if (isCentered(item)) {
+          recenter(dropdown, item)
+        }
+      })
     })
   }
 }
+
+function recenter(dropdown: HTMLElement, item: HTMLElement) {
```

Comment:
After some thought I've decided against extracting this function because it's not clear what does it center and what are it's assumptions. There are many different ways to center something so having a _generic_ function to make elements be center is very hard